tuf/ngclient/updater.py

```diff
@@ -415,9 +415,10 @@ def _preorder_depth_first_walk(
                 logger.debug("Skipping visited current role %s", role_name)
                 continue
 
-            # The metadata for 'role_name' must be downloaded/updated before
-            # its targets, delegations, and child roles can be inspected.
-            self._load_targets(role_name, parent_role)
+            # The metadata for 'role_name' must be loaded
+            # before its targets and delegations can be inspected.
+            if not self._trusted_set.get(role_name):
```

Comment:
more pythonic IMO:

```suggestion
            if role_name not in self._trusted_set:
```

Reply:
yeah, very pythonic :) it is updated now.